
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://monarch-app.monarchinitiative.org/docs/Usage/CLI/">
      
      
        <link rel="prev" href="../">
      
      
        <link rel="next" href="../../Data-Model/">
      
      
      <link rel="icon" href="../../images/favicon.ico">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.14">
    
    
      
        <title>CLI - Monarch KG Tools</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.342714a4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300i,400,400i,700,700i%7CSource+Sans+Pro+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Source Sans Pro";--md-code-font:"Source Sans Pro Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../assets/_mkdocstrings.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#monarch" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Monarch KG Tools" class="md-header__button md-logo" aria-label="Monarch KG Tools" data-md-component="logo">
      
  <img src="../../images/monarch-initiative.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Monarch KG Tools
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              CLI
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 2c-1.82 0-3.53.5-5 1.35C8 5.08 10 8.3 10 12s-2 6.92-5 8.65C6.47 21.5 8.18 22 10 22a10 10 0 0 0 10-10A10 10 0 0 0 10 2"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/monarch-initiative/monarch-app/tree/main/backend" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    monarch-app/backend
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../.." class="md-tabs__link">
          
  
  
  monarch-py

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../release-process/" class="md-tabs__link">
        
  
  
    
  
  Release Process

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../FastAPI/" class="md-tabs__link">
          
  
  
  FastAPI

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Monarch KG Tools" class="md-nav__button md-logo" aria-label="Monarch KG Tools" data-md-component="logo">
      
  <img src="../../images/monarch-initiative.png" alt="logo">

    </a>
    Monarch KG Tools
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/monarch-initiative/monarch-app/tree/main/backend" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    monarch-app/backend
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
    
  
  
  
    
    
    
    
      
        
        
      
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" checked>
        
          
          <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    monarch-py
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_1">
            <span class="md-nav__icon md-icon"></span>
            monarch-py
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Usage
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    CLI
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    CLI
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#monarch-test" class="md-nav__link">
    <span class="md-ellipsis">
      monarch test
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-schema" class="md-nav__link">
    <span class="md-ellipsis">
      monarch schema
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch entity
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch associations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-multi-entity-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch multi-entity-associations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-search" class="md-nav__link">
    <span class="md-ellipsis">
      monarch search
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-autocomplete" class="md-nav__link">
    <span class="md-ellipsis">
      monarch autocomplete
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-histopheno" class="md-nav__link">
    <span class="md-ellipsis">
      monarch histopheno
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-association-counts" class="md-nav__link">
    <span class="md-ellipsis">
      monarch association-counts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-association-table" class="md-nav__link">
    <span class="md-ellipsis">
      monarch association-table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-mappings" class="md-nav__link">
    <span class="md-ellipsis">
      monarch mappings
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-compare" class="md-nav__link">
    <span class="md-ellipsis">
      monarch compare
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-releases" class="md-nav__link">
    <span class="md-ellipsis">
      monarch releases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-release" class="md-nav__link">
    <span class="md-ellipsis">
      monarch release
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-solr" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr
    </span>
  </a>
  
    <nav class="md-nav" aria-label="monarch solr">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#monarch-solr-start" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr start
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-stop" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr stop
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-status" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr status
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-download" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr download
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr entity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr associations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-multi-entity-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr multi-entity-associations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-search" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr search
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-autocomplete" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr autocomplete
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-histopheno" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr histopheno
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-association-counts" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr association-counts
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-association-table" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr association-table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-mappings" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr mappings
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-sql" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql
    </span>
  </a>
  
    <nav class="md-nav" aria-label="monarch sql">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#monarch-sql-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql entity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-sql-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql associations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Data-Model/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Data Model
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_5" >
        
          
          <label class="md-nav__link" for="__nav_1_5" id="__nav_1_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Interfaces
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_5">
            <span class="md-nav__icon md-icon"></span>
            Interfaces
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Interfaces/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Interfaces/Entity/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Entity
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Interfaces/Association/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Association
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Interfaces/Query/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Query
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Interfaces/Search/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Search
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_1_6" >
        
          
          <label class="md-nav__link" for="__nav_1_6" id="__nav_1_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Implementations
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_1_6">
            <span class="md-nav__icon md-icon"></span>
            Implementations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Implementations/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Implementations/Solr/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Solr
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Implementations/SQL/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    SQL
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../release-process/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Release Process
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    FastAPI
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            FastAPI
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../FastAPI/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3_2" >
        
          
          <label class="md-nav__link" for="__nav_3_2" id="__nav_3_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Endpoints
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_3_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3_2">
            <span class="md-nav__icon md-icon"></span>
            Endpoints
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../FastAPI/Endpoints/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Monarch API Endpoints
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#monarch-test" class="md-nav__link">
    <span class="md-ellipsis">
      monarch test
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-schema" class="md-nav__link">
    <span class="md-ellipsis">
      monarch schema
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch entity
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch associations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-multi-entity-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch multi-entity-associations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-search" class="md-nav__link">
    <span class="md-ellipsis">
      monarch search
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-autocomplete" class="md-nav__link">
    <span class="md-ellipsis">
      monarch autocomplete
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-histopheno" class="md-nav__link">
    <span class="md-ellipsis">
      monarch histopheno
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-association-counts" class="md-nav__link">
    <span class="md-ellipsis">
      monarch association-counts
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-association-table" class="md-nav__link">
    <span class="md-ellipsis">
      monarch association-table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-mappings" class="md-nav__link">
    <span class="md-ellipsis">
      monarch mappings
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-compare" class="md-nav__link">
    <span class="md-ellipsis">
      monarch compare
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-releases" class="md-nav__link">
    <span class="md-ellipsis">
      monarch releases
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-release" class="md-nav__link">
    <span class="md-ellipsis">
      monarch release
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-solr" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr
    </span>
  </a>
  
    <nav class="md-nav" aria-label="monarch solr">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#monarch-solr-start" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr start
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-stop" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr stop
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-status" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr status
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-download" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr download
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr entity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr associations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-multi-entity-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr multi-entity-associations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-search" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr search
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-autocomplete" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr autocomplete
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-histopheno" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr histopheno
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-association-counts" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr association-counts
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-association-table" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr association-table
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-solr-mappings" class="md-nav__link">
    <span class="md-ellipsis">
      monarch solr mappings
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monarch-sql" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql
    </span>
  </a>
  
    <nav class="md-nav" aria-label="monarch sql">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#monarch-sql-entity" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql entity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#monarch-sql-associations" class="md-nav__link">
    <span class="md-ellipsis">
      monarch sql associations
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="monarch"><code>monarch</code></h1>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-v, --version</code>: Show the currently installed version</li>
<li><code>-q, --quiet</code>: Set log level to warning</li>
<li><code>-d, --debug</code>: Set log level to debug</li>
<li><code>--install-completion</code>: Install completion for the current shell.</li>
<li><code>--show-completion</code>: Show completion for the current shell, to copy it or customize the installation.</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<p><strong>Commands</strong>:</p>
<ul>
<li><code>test</code>: Test the CLI</li>
<li><code>schema</code>: Print the linkml schema for the data model</li>
<li><code>entity</code>: Retrieve an entity by ID</li>
<li><code>associations</code>: Paginate through associations</li>
<li><code>multi-entity-associations</code>: Paginate through associations for multiple...</li>
<li><code>search</code>: Search for entities</li>
<li><code>autocomplete</code>: Return entity autcomplete matches for a...</li>
<li><code>histopheno</code>: Retrieve the histopheno associations for a...</li>
<li><code>association-counts</code>: Retrieve the association counts for a...</li>
<li><code>association-table</code></li>
<li><code>mappings</code></li>
<li><code>compare</code>: Compare two sets of phenotypes using...</li>
<li><code>releases</code>: List all available releases of the Monarch...</li>
<li><code>release</code>: Retrieve metadata for a specific release</li>
<li><code>solr</code></li>
<li><code>sql</code></li>
</ul>
<h2 id="monarch-test"><code>monarch test</code></h2>
<p>Test the CLI</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span><span class="nb">test</span><span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-schema"><code>monarch schema</code></h2>
<p>Print the linkml schema for the data model</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>schema<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-entity"><code>monarch entity</code></h2>
<p>Retrieve an entity by ID</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>entity<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The identifier of the entity to be retrieved  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --extra</code>: Include extra fields in the output (association_counts and node_hierarchy)</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-associations"><code>monarch associations</code></h2>
<p>Paginate through associations</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>associations<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-c, --category [biolink:PairwiseGeneToGeneInteraction|biolink:GeneToExpressionSiteAssociation|biolink:GeneToGeneHomologyAssociation|biolink:Association|biolink:MacromolecularMachineToBiologicalProcessAssociation|biolink:MacromolecularMachineToMolecularActivityAssociation|biolink:MacromolecularMachineToCellularComponentAssociation|biolink:GeneToPhenotypicFeatureAssociation|biolink:GenotypeToPhenotypicFeatureAssociation|biolink:VariantToPhenotypicFeatureAssociation|biolink:DiseaseToPhenotypicFeatureAssociation|biolink:GeneToPathwayAssociation|biolink:VariantToGeneAssociation|biolink:CaseToPhenotypicFeatureAssociation|biolink:GenotypeToGeneAssociation|biolink:GenotypeToVariantAssociation|biolink:ChemicalEntityToPathwayAssociation|biolink:VariantToDiseaseAssociation|biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation|biolink:GenotypeToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation|biolink:CorrelatedGeneToDiseaseAssociation|biolink:CaseToDiseaseAssociation|biolink:CaseToGeneAssociation|biolink:CausalGeneToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToLocationAssociation|biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation]</code>: Category to get associations for</li>
<li><code>-s, --subject TEXT</code>: Subject ID to get associations for</li>
<li><code>-p, --predicate [biolink:interacts_with|biolink:expressed_in|biolink:has_phenotype|biolink:orthologous_to|biolink:enables|biolink:actively_involved_in|biolink:related_to|biolink:located_in|biolink:subclass_of|biolink:participates_in|biolink:acts_upstream_of_or_within|biolink:is_sequence_variant_of|biolink:is_active_in|biolink:has_sequence_variant|biolink:part_of|biolink:causes|biolink:treats_or_applied_or_studied_to_treat|biolink:homologous_to|biolink:contributes_to|biolink:model_of|biolink:has_mode_of_inheritance|biolink:has_disease|biolink:has_gene|biolink:gene_associated_with_condition|biolink:associated_with_increased_likelihood_of|biolink:colocalizes_with|biolink:same_as|biolink:acts_upstream_of|biolink:genetically_associated_with|biolink:disease_has_location|biolink:acts_upstream_of_or_within_positive_effect|biolink:ameliorates_condition|biolink:acts_upstream_of_positive_effect|biolink:acts_upstream_of_negative_effect|biolink:acts_upstream_of_or_within_negative_effect|biolink:has_participant|biolink:preventative_for_condition|biolink:disrupts|biolink:caused_by|biolink:contraindicated_in]</code>: Predicate ID to get associations for</li>
<li><code>-o, --object TEXT</code>: Object ID to get associations for</li>
<li><code>-e, --entity TEXT</code>: Entity (subject or object) ID to get associations for</li>
<li><code>-d, --direct</code>: Whether to exclude associations with subject/object as ancestors</li>
<li><code>-C, --compact</code>: Whether to return a compact representation of the associations</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-multi-entity-associations"><code>monarch multi-entity-associations</code></h2>
<p>Paginate through associations for multiple entities</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>multi-entity-associations<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --entity TEXT</code>: Comma-separated list of entities</li>
<li><code>-c, --counterpart-category TEXT</code>: A comma-separated list of counterpart categories</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-search"><code>monarch search</code></h2>
<p>Search for entities</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>search<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --query TEXT</code>: [default: :*]</li>
<li><code>-c, --category [biolink:Gene|biolink:SequenceVariant|biolink:PhenotypicFeature|biolink:Genotype|biolink:AnatomicalEntity|biolink:NamedThing|biolink:BiologicalProcess|biolink:Disease|biolink:Protein|biolink:Pathway|biolink:ChemicalEntity|biolink:MolecularActivity|biolink:Case|biolink:CellularComponent|biolink:Cell|biolink:OrganismTaxon|biolink:MolecularEntity|biolink:LifeStage]</code></li>
<li><code>-t, --in-taxon-label TEXT</code></li>
<li><code>-ff, --facet-fields TEXT</code></li>
<li><code>-fq, --facet-queries TEXT</code></li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-autocomplete"><code>monarch autocomplete</code></h2>
<p>Return entity autcomplete matches for a query string</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>autocomplete<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>Q
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>Q</code>: Query string to autocomplete against  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-histopheno"><code>monarch histopheno</code></h2>
<p>Retrieve the histopheno associations for a given subject</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>histopheno<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>SUBJECT
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>SUBJECT</code>: The subject of the association  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-association-counts"><code>monarch association-counts</code></h2>
<p>Retrieve the association counts for a given entity</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>association-counts<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The entity to get association counts for  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-association-table"><code>monarch association-table</code></h2>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>association-table<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID<span class="w"> </span>CATEGORY:<span class="o">{</span>biolink:PairwiseGeneToGeneInteraction<span class="p">|</span>biolink:GeneToExpressionSiteAssociation<span class="p">|</span>biolink:GeneToGeneHomologyAssociation<span class="p">|</span>biolink:Association<span class="p">|</span>biolink:MacromolecularMachineToBiologicalProcessAssociation<span class="p">|</span>biolink:MacromolecularMachineToMolecularActivityAssociation<span class="p">|</span>biolink:MacromolecularMachineToCellularComponentAssociation<span class="p">|</span>biolink:GeneToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToPhenotypicFeatureAssociation<span class="p">|</span>biolink:VariantToPhenotypicFeatureAssociation<span class="p">|</span>biolink:DiseaseToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GeneToPathwayAssociation<span class="p">|</span>biolink:VariantToGeneAssociation<span class="p">|</span>biolink:CaseToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToGeneAssociation<span class="p">|</span>biolink:GenotypeToVariantAssociation<span class="p">|</span>biolink:ChemicalEntityToPathwayAssociation<span class="p">|</span>biolink:VariantToDiseaseAssociation<span class="p">|</span>biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToDiseaseAssociation<span class="p">|</span>biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation<span class="p">|</span>biolink:CorrelatedGeneToDiseaseAssociation<span class="p">|</span>biolink:CaseToDiseaseAssociation<span class="p">|</span>biolink:CaseToGeneAssociation<span class="p">|</span>biolink:CausalGeneToDiseaseAssociation<span class="p">|</span>biolink:DiseaseOrPhenotypicFeatureToLocationAssociation<span class="p">|</span>biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation<span class="o">}</span>
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The entity to get associations for  [required]</li>
<li><code>CATEGORY:{biolink:PairwiseGeneToGeneInteraction|biolink:GeneToExpressionSiteAssociation|biolink:GeneToGeneHomologyAssociation|biolink:Association|biolink:MacromolecularMachineToBiologicalProcessAssociation|biolink:MacromolecularMachineToMolecularActivityAssociation|biolink:MacromolecularMachineToCellularComponentAssociation|biolink:GeneToPhenotypicFeatureAssociation|biolink:GenotypeToPhenotypicFeatureAssociation|biolink:VariantToPhenotypicFeatureAssociation|biolink:DiseaseToPhenotypicFeatureAssociation|biolink:GeneToPathwayAssociation|biolink:VariantToGeneAssociation|biolink:CaseToPhenotypicFeatureAssociation|biolink:GenotypeToGeneAssociation|biolink:GenotypeToVariantAssociation|biolink:ChemicalEntityToPathwayAssociation|biolink:VariantToDiseaseAssociation|biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation|biolink:GenotypeToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation|biolink:CorrelatedGeneToDiseaseAssociation|biolink:CaseToDiseaseAssociation|biolink:CaseToGeneAssociation|biolink:CausalGeneToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToLocationAssociation|biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation}</code>: The association category to get associations for, ex. biolink:GeneToPhenotypicFeatureAssociation  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --query TEXT</code></li>
<li><code>-t, --traverse-orthologs</code>: Whether to traverse orthologs when getting associations</li>
<li><code>-s, --sort TEXT</code></li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-mappings"><code>monarch mappings</code></h2>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>mappings<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --entity-id TEXT</code>: entity ID to get mappings for</li>
<li><code>-s, --subject-id TEXT</code>: subject ID to get mappings for</li>
<li><code>-p, --predicate-id [skos:exactMatch|skos:closeMatch|skos:broadMatch]</code>: predicate ID to get mappings for</li>
<li><code>-o, --object-id TEXT</code>: object ID to get mappings for</li>
<li><code>-m, --mapping-justification TEXT</code>: mapping justification to get mappings for</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-compare"><code>monarch compare</code></h2>
<p>Compare two sets of phenotypes using semantic similarity via SemSimian</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>compare<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>SUBJECTS<span class="w"> </span>OBJECTS
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>SUBJECTS</code>: Comma separated list of subjects to compare  [required]</li>
<li><code>OBJECTS</code>: Comma separated list of objects to compare  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-m, --metric [ancestor_information_content|jaccard_similarity|phenodigm_score]</code>: The metric to use for comparison  [default: ancestor_information_content]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-releases"><code>monarch releases</code></h2>
<p>List all available releases of the Monarch Knowledge Graph</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>releases<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--dev</code>: Get dev releases of the KG (default is False)</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 0]</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-release"><code>monarch release</code></h2>
<p>Retrieve metadata for a specific release</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>release<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>RELEASE_VER
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>RELEASE_VER</code>: The release version to get metadata for  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-solr"><code>monarch solr</code></h2>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --quiet</code>: Set log level to warning</li>
<li><code>-d, --debug</code>: Set log level to debug</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<p><strong>Commands</strong>:</p>
<ul>
<li><code>start</code>: Starts a local Solr container.</li>
<li><code>stop</code>: Stops the local Solr container.</li>
<li><code>status</code>: Checks the status of the local Solr...</li>
<li><code>download</code>: Download the Monarch Solr KG.</li>
<li><code>entity</code>: Retrieve an entity by ID</li>
<li><code>associations</code>: Paginate through associations</li>
<li><code>multi-entity-associations</code>: Paginate through associations for multiple...</li>
<li><code>search</code>: Search for entities</li>
<li><code>autocomplete</code>: Return entity autcomplete matches for a...</li>
<li><code>histopheno</code>: Retrieve the histopheno associations for a...</li>
<li><code>association-counts</code>: Retrieve the association counts for a...</li>
<li><code>association-table</code></li>
<li><code>mappings</code></li>
</ul>
<h3 id="monarch-solr-start"><code>monarch solr start</code></h3>
<p>Starts a local Solr container.</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>start<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--update / --no-update</code>: [default: no-update]</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-stop"><code>monarch solr stop</code></h3>
<p>Stops the local Solr container.</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>stop<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-status"><code>monarch solr status</code></h3>
<p>Checks the status of the local Solr container.</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>status<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-download"><code>monarch solr download</code></h3>
<p>Download the Monarch Solr KG.</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>download<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span><span class="o">[</span>VERSION<span class="o">]</span>
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>[VERSION]</code>: The version of the Solr KG to download (latest, dev, or a specific version)  [default: latest]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>--overwrite</code>: Overwrite the existing Solr KG if it exists</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-entity"><code>monarch solr entity</code></h3>
<p>Retrieve an entity by ID</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>entity<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The identifier of the entity to be retrieved  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --extra</code>: Include extra fields in the output (association_counts and node_hierarchy)</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-associations"><code>monarch solr associations</code></h3>
<p>Paginate through associations</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>associations<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-c, --category [biolink:PairwiseGeneToGeneInteraction|biolink:GeneToExpressionSiteAssociation|biolink:GeneToGeneHomologyAssociation|biolink:Association|biolink:MacromolecularMachineToBiologicalProcessAssociation|biolink:MacromolecularMachineToMolecularActivityAssociation|biolink:MacromolecularMachineToCellularComponentAssociation|biolink:GeneToPhenotypicFeatureAssociation|biolink:GenotypeToPhenotypicFeatureAssociation|biolink:VariantToPhenotypicFeatureAssociation|biolink:DiseaseToPhenotypicFeatureAssociation|biolink:GeneToPathwayAssociation|biolink:VariantToGeneAssociation|biolink:CaseToPhenotypicFeatureAssociation|biolink:GenotypeToGeneAssociation|biolink:GenotypeToVariantAssociation|biolink:ChemicalEntityToPathwayAssociation|biolink:VariantToDiseaseAssociation|biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation|biolink:GenotypeToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation|biolink:CorrelatedGeneToDiseaseAssociation|biolink:CaseToDiseaseAssociation|biolink:CaseToGeneAssociation|biolink:CausalGeneToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToLocationAssociation|biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation]</code>: Category to get associations for</li>
<li><code>-s, --subject TEXT</code>: Subject ID to get associations for</li>
<li><code>-p, --predicate [biolink:interacts_with|biolink:expressed_in|biolink:has_phenotype|biolink:orthologous_to|biolink:enables|biolink:actively_involved_in|biolink:related_to|biolink:located_in|biolink:subclass_of|biolink:participates_in|biolink:acts_upstream_of_or_within|biolink:is_sequence_variant_of|biolink:is_active_in|biolink:has_sequence_variant|biolink:part_of|biolink:causes|biolink:treats_or_applied_or_studied_to_treat|biolink:homologous_to|biolink:contributes_to|biolink:model_of|biolink:has_mode_of_inheritance|biolink:has_disease|biolink:has_gene|biolink:gene_associated_with_condition|biolink:associated_with_increased_likelihood_of|biolink:colocalizes_with|biolink:same_as|biolink:acts_upstream_of|biolink:genetically_associated_with|biolink:disease_has_location|biolink:acts_upstream_of_or_within_positive_effect|biolink:ameliorates_condition|biolink:acts_upstream_of_positive_effect|biolink:acts_upstream_of_negative_effect|biolink:acts_upstream_of_or_within_negative_effect|biolink:has_participant|biolink:preventative_for_condition|biolink:disrupts|biolink:caused_by|biolink:contraindicated_in]</code>: Predicate ID to get associations for</li>
<li><code>-o, --object TEXT</code>: Object ID to get associations for</li>
<li><code>-e, --entity TEXT</code>: Entity (subject or object) ID to get associations for</li>
<li><code>-d, --direct</code>: Whether to exclude associations with subject/object as ancestors</li>
<li><code>-C, --compact</code>: Whether to return a compact representation of the associations</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-multi-entity-associations"><code>monarch solr multi-entity-associations</code></h3>
<p>Paginate through associations for multiple entities</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>multi-entity-associations<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --entity TEXT</code>: Comma-separated list of entities</li>
<li><code>-c, --counterpart-category TEXT</code>: A comma-separated list of counterpart categories</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-search"><code>monarch solr search</code></h3>
<p>Search for entities</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>search<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --query TEXT</code>: [default: :*]</li>
<li><code>-c, --category [biolink:Gene|biolink:SequenceVariant|biolink:PhenotypicFeature|biolink:Genotype|biolink:AnatomicalEntity|biolink:NamedThing|biolink:BiologicalProcess|biolink:Disease|biolink:Protein|biolink:Pathway|biolink:ChemicalEntity|biolink:MolecularActivity|biolink:Case|biolink:CellularComponent|biolink:Cell|biolink:OrganismTaxon|biolink:MolecularEntity|biolink:LifeStage]</code></li>
<li><code>-t, --in-taxon-label TEXT</code></li>
<li><code>-ff, --facet-fields TEXT</code></li>
<li><code>-fq, --facet-queries TEXT</code></li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-autocomplete"><code>monarch solr autocomplete</code></h3>
<p>Return entity autcomplete matches for a query string</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>autocomplete<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>Q
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>Q</code>: Query string to autocomplete against  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-histopheno"><code>monarch solr histopheno</code></h3>
<p>Retrieve the histopheno associations for a given subject</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>histopheno<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>SUBJECT
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>SUBJECT</code>: The subject of the association  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-association-counts"><code>monarch solr association-counts</code></h3>
<p>Retrieve the association counts for a given entity</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>association-counts<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The entity to get association counts for  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-association-table"><code>monarch solr association-table</code></h3>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>association-table<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID<span class="w"> </span>CATEGORY:<span class="o">{</span>biolink:PairwiseGeneToGeneInteraction<span class="p">|</span>biolink:GeneToExpressionSiteAssociation<span class="p">|</span>biolink:GeneToGeneHomologyAssociation<span class="p">|</span>biolink:Association<span class="p">|</span>biolink:MacromolecularMachineToBiologicalProcessAssociation<span class="p">|</span>biolink:MacromolecularMachineToMolecularActivityAssociation<span class="p">|</span>biolink:MacromolecularMachineToCellularComponentAssociation<span class="p">|</span>biolink:GeneToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToPhenotypicFeatureAssociation<span class="p">|</span>biolink:VariantToPhenotypicFeatureAssociation<span class="p">|</span>biolink:DiseaseToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GeneToPathwayAssociation<span class="p">|</span>biolink:VariantToGeneAssociation<span class="p">|</span>biolink:CaseToPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToGeneAssociation<span class="p">|</span>biolink:GenotypeToVariantAssociation<span class="p">|</span>biolink:ChemicalEntityToPathwayAssociation<span class="p">|</span>biolink:VariantToDiseaseAssociation<span class="p">|</span>biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation<span class="p">|</span>biolink:GenotypeToDiseaseAssociation<span class="p">|</span>biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation<span class="p">|</span>biolink:CorrelatedGeneToDiseaseAssociation<span class="p">|</span>biolink:CaseToDiseaseAssociation<span class="p">|</span>biolink:CaseToGeneAssociation<span class="p">|</span>biolink:CausalGeneToDiseaseAssociation<span class="p">|</span>biolink:DiseaseOrPhenotypicFeatureToLocationAssociation<span class="p">|</span>biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation<span class="o">}</span>
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The entity to get associations for  [required]</li>
<li><code>CATEGORY:{biolink:PairwiseGeneToGeneInteraction|biolink:GeneToExpressionSiteAssociation|biolink:GeneToGeneHomologyAssociation|biolink:Association|biolink:MacromolecularMachineToBiologicalProcessAssociation|biolink:MacromolecularMachineToMolecularActivityAssociation|biolink:MacromolecularMachineToCellularComponentAssociation|biolink:GeneToPhenotypicFeatureAssociation|biolink:GenotypeToPhenotypicFeatureAssociation|biolink:VariantToPhenotypicFeatureAssociation|biolink:DiseaseToPhenotypicFeatureAssociation|biolink:GeneToPathwayAssociation|biolink:VariantToGeneAssociation|biolink:CaseToPhenotypicFeatureAssociation|biolink:GenotypeToGeneAssociation|biolink:GenotypeToVariantAssociation|biolink:ChemicalEntityToPathwayAssociation|biolink:VariantToDiseaseAssociation|biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation|biolink:GenotypeToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation|biolink:CorrelatedGeneToDiseaseAssociation|biolink:CaseToDiseaseAssociation|biolink:CaseToGeneAssociation|biolink:CausalGeneToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToLocationAssociation|biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation}</code>: The association category to get associations for, ex. biolink:GeneToPhenotypicFeatureAssociation  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --query TEXT</code></li>
<li><code>-t, --traverse-orthologs</code>: Whether to traverse orthologs when getting associations</li>
<li><code>-s, --sort TEXT</code></li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-solr-mappings"><code>monarch solr mappings</code></h3>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>solr<span class="w"> </span>mappings<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --entity-id TEXT</code>: entity ID to get mappings for</li>
<li><code>-s, --subject-id TEXT</code>: subject ID to get mappings for</li>
<li><code>-p, --predicate-id [skos:exactMatch|skos:closeMatch|skos:broadMatch]</code>: predicate ID to get mappings for</li>
<li><code>-o, --object-id TEXT</code>: object ID to get mappings for</li>
<li><code>-m, --mapping-justification TEXT</code>: mapping justification to get mappings for</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h2 id="monarch-sql"><code>monarch sql</code></h2>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>sql<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>COMMAND<span class="w"> </span><span class="o">[</span>ARGS<span class="o">]</span>...
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-q, --quiet</code>: Set log level to warning</li>
<li><code>-d, --debug</code>: Set log level to debug</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<p><strong>Commands</strong>:</p>
<ul>
<li><code>entity</code>: Retrieve an entity by ID</li>
<li><code>associations</code>: Paginate through associations</li>
</ul>
<h3 id="monarch-sql-entity"><code>monarch sql entity</code></h3>
<p>Retrieve an entity by ID</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>sql<span class="w"> </span>entity<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span><span class="w"> </span>ENTITY_ID
</code></pre></div>
<p><strong>Arguments</strong>:</p>
<ul>
<li><code>ENTITY_ID</code>: The identifier of the entity to be retrieved  [required]</li>
</ul>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-e, --extra</code>: Include extra fields in the output (association_counts and node_hierarchy)  [required]</li>
<li><code>-u, --update</code>: Whether to re-download the Monarch KG  [required]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>
<h3 id="monarch-sql-associations"><code>monarch sql associations</code></h3>
<p>Paginate through associations</p>
<p><strong>Usage</strong>:</p>
<div class="highlight"><pre><span></span><code><span class="gp">$ </span>monarch<span class="w"> </span>sql<span class="w"> </span>associations<span class="w"> </span><span class="o">[</span>OPTIONS<span class="o">]</span>
</code></pre></div>
<p><strong>Options</strong>:</p>
<ul>
<li><code>-c, --category [biolink:PairwiseGeneToGeneInteraction|biolink:GeneToExpressionSiteAssociation|biolink:GeneToGeneHomologyAssociation|biolink:Association|biolink:MacromolecularMachineToBiologicalProcessAssociation|biolink:MacromolecularMachineToMolecularActivityAssociation|biolink:MacromolecularMachineToCellularComponentAssociation|biolink:GeneToPhenotypicFeatureAssociation|biolink:GenotypeToPhenotypicFeatureAssociation|biolink:VariantToPhenotypicFeatureAssociation|biolink:DiseaseToPhenotypicFeatureAssociation|biolink:GeneToPathwayAssociation|biolink:VariantToGeneAssociation|biolink:CaseToPhenotypicFeatureAssociation|biolink:GenotypeToGeneAssociation|biolink:GenotypeToVariantAssociation|biolink:ChemicalEntityToPathwayAssociation|biolink:VariantToDiseaseAssociation|biolink:ChemicalEntityToDiseaseOrPhenotypicFeatureAssociation|biolink:GenotypeToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToGeneticInheritanceAssociation|biolink:CorrelatedGeneToDiseaseAssociation|biolink:CaseToDiseaseAssociation|biolink:CaseToGeneAssociation|biolink:CausalGeneToDiseaseAssociation|biolink:DiseaseOrPhenotypicFeatureToLocationAssociation|biolink:ChemicalOrDrugOrTreatmentToDiseaseOrPhenotypicFeatureAssociation]</code>: Category to get associations for</li>
<li><code>-s, --subject TEXT</code>: Subject ID to get associations for</li>
<li><code>-p, --predicate [biolink:interacts_with|biolink:expressed_in|biolink:has_phenotype|biolink:orthologous_to|biolink:enables|biolink:actively_involved_in|biolink:related_to|biolink:located_in|biolink:subclass_of|biolink:participates_in|biolink:acts_upstream_of_or_within|biolink:is_sequence_variant_of|biolink:is_active_in|biolink:has_sequence_variant|biolink:part_of|biolink:causes|biolink:treats_or_applied_or_studied_to_treat|biolink:homologous_to|biolink:contributes_to|biolink:model_of|biolink:has_mode_of_inheritance|biolink:has_disease|biolink:has_gene|biolink:gene_associated_with_condition|biolink:associated_with_increased_likelihood_of|biolink:colocalizes_with|biolink:same_as|biolink:acts_upstream_of|biolink:genetically_associated_with|biolink:disease_has_location|biolink:acts_upstream_of_or_within_positive_effect|biolink:ameliorates_condition|biolink:acts_upstream_of_positive_effect|biolink:acts_upstream_of_negative_effect|biolink:acts_upstream_of_or_within_negative_effect|biolink:has_participant|biolink:preventative_for_condition|biolink:disrupts|biolink:caused_by|biolink:contraindicated_in]</code>: Predicate ID to get associations for</li>
<li><code>-o, --object TEXT</code>: Object ID to get associations for</li>
<li><code>-e, --entity TEXT</code>: Entity (subject or object) ID to get associations for</li>
<li><code>-d, --direct</code>: Whether to exclude associations with subject/object as ancestors</li>
<li><code>-C, --compact</code>: Whether to return a compact representation of the associations</li>
<li><code>-l, --limit INTEGER</code>: The number of results to return  [default: 20]</li>
<li><code>--offset INTEGER</code>: The offset of the first result to be retrieved  [default: 0]</li>
<li><code>-f, --format [json|yaml|tsv|table]</code>: The format of the output (json, yaml, tsv, table)  [default: json]</li>
<li><code>-O, --output TEXT</code>: Path to file to write command output (stdout if not specified)</li>
<li><code>--help</code>: Show this message and exit.</li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://monarchinitiative.org" target="_blank" rel="noopener" title="monarchinitiative.org" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M575.8 255.5c0 18-15 32.1-32 32.1h-32l.7 160.2c0 2.7-.2 5.4-.5 8.1v16.2c0 22.1-17.9 40-40 40h-16c-1.1 0-2.2 0-3.3-.1-1.4.1-2.8.1-4.2.1L416 512h-24c-22.1 0-40-17.9-40-40v-88c0-17.7-14.3-32-32-32h-64c-17.7 0-32 14.3-32 32v88c0 22.1-17.9 40-40 40h-55.9c-1.5 0-3-.1-4.5-.2-1.2.1-2.4.2-3.6.2h-16c-22.1 0-40-17.9-40-40V360c0-.9 0-1.9.1-2.8v-69.7h-32c-18 0-32-14-32-32.1 0-9 3-17 10-24L266.4 8c7-7 15-8 22-8s15 2 21 7l255.4 224.5c8 7 12 15 11 24"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://github.com/monarch-initiative/" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 480 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1M480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2m-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3m-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.expand", "navigation.instant", "navigation.tracking", "navigation.tabs", "navigation.tabs.sticky"], "search": "../../assets/javascripts/workers/search.d50fe291.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.13a4f30d.min.js"></script>
      
    
  </body>
</html>